### PR TITLE
DDB Enhanced: updates README with bean examples

### PR DESCRIPTION
**Pull request by cenedhryn**
## Description
Adds sections for the StaticTableSchema and BeanTableSchema, including usage of
- AttributeConverterProviders
- AttributeConverters
- Flattning and extending beans
- Ignoring and including bean attributes

### CODE REVIEW

File: services-custom/dynamodb-enhanced/README.md [hunk 1]
```diff
@@ -291,32 +296,205 @@ Using composition:
...
+
+    public String getCreatedDate() { return this.createdDate; }
+    public void setCreatedDate(String createdDate) { this.createdDate = createdDate;}
+}
+```
+You can flatten as many different eligible classes as you like using the flatten annotation.
+The only constraints are that attributes must not have the same name when they are being rolled
+together, and there must never be more than one partition key, sort key or table name.
+
+### Control how attributes are converted
```
Comment (bmaizels) [inline services-custom/dynamodb-enhanced/README.md]: Same comment as before, I think we should organize by use-case and then a description of how to do it in the BeanTableSchema and then the StaticTableSchema

File: services-custom/dynamodb-enhanced/README.md [hunk 2]
```diff
@@ -291,32 +296,205 @@ Using composition:
...
+    .attributeConverterProviders(converterProvider1, converterProvider2)
+    .build();
+```
+Note that if you supply your own chain of attribute converter providers, you will override
+the default converter provider and must therefore include it in the chain if you wish to
+use its attribute converters.
+
+#### Override the mapping of a single attribute
+Supply an AttributeConverter when creating the attribute to directly override any
+converters provided by the table schema AttributeConverterProviders:
```
[the remark this answers is not in the document]
Comment (cenedhryn) [reply services-custom/dynamodb-enhanced/README.md]: ok

File: services-custom/dynamodb-enhanced/README.md [hunk 3]
```diff
@@ -291,32 +296,205 @@ Using composition:
...
+using the @DynamoDbAttribute annotation:
+```java
+private String internalKey;
+
+@DynamoDbAttribute("renamedInternalKey")
+public String getInternalKey() { return this.internalKey; }
+public void setInternalKey(String internalKey) { return this.internalKey = internalKey;}
+```
+
+### Flat map attributes from another class
```
[the remark this answers is not in the document]
Comment (cenedhryn) [reply services-custom/dynamodb-enhanced/README.md]: Yes, I rearranged it and its clearer, consistent and less repetitive.

File: services-custom/dynamodb-enhanced/README.md [hunk 4]
```diff
@@ -248,14 +246,143 @@ Or using a StaticTableSchema:
...
+public class Customer {
+
+}
+```
+
+Chain of converter providers ending with the default (least priority):
+```java
+@DynamoDbBean(converterProviders = {
+   converterProvider1.class, 
+   converterProvider2.class,
```
Comment (bmaizels) [inline services-custom/dynamodb-enhanced/README.md]: Capital C?

File: services-custom/dynamodb-enhanced/README.md [hunk 5]
```diff
@@ -270,53 +397,96 @@ public abstract class GenericRecord {
...
+    public String getId() { return this.id; }
+    public void setId(String id) { this.id = id;}
+
+    public String getCreatedDate() { return this.createdDate; }
+    public void setCreatedDate(String createdDate) { this.createdDate = createdDate;}
+}
+```
+You can flatten as many different eligible classes as you like using the flatten annotation.
+The only constraints are that attributes must not have the same name when they are being rolled
+together, and there must never be more than one partition key, sort key or table name.
```
Comment (bmaizels) [inline services-custom/dynamodb-enhanced/README.md]: I think we should move this text underneath the StaticTableSchema example.

File: services-custom/dynamodb-enhanced/README.md (hunk 4 above)
Comment (cenedhryn) [reply services-custom/dynamodb-enhanced/README.md]: :-)

**Pull request merged**